eos/scattering/parametric-gmkprdey2011_TEST.cc

```diff
@@ -0,0 +1,138 @@
+                    std::make_pair(  3.140443,  eps),       // del_D0(s =  4.0)
+                };
+                TEST_CHECK_DIAGNOSTICS(diagnostics, reference);
+
+                TEST_CHECK_NEARLY_EQUAL( amp.scattering_amplitude(0.9 , 0 , "0").real() , -0.449485 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.scattering_amplitude(0.9 , 0 , "0").imag() ,  0.790851 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.scattering_amplitude(0.9 , 1 , "1").real() , -0.498251 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.scattering_amplitude(0.9 , 1 , "1").imag() ,  0.290285 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.scattering_amplitude(0.9 , 2 , "0").real() ,  0.215659 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.scattering_amplitude(0.9 , 2 , "0").imag() ,  0.038253 ,  eps);
+
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(0.55  , 1 , "1").real()       ,  2.542132 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(0.55  , 1 , "1").imag()       ,  5.375608 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(0.9   , 1 , "1").real()       , -1.760142 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(0.9   , 1 , "1").imag()       ,  1.025474 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(9.0   , 1 , "1").real()       , -0.081634 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(9.0   , 1 , "1").imag()       ,  0.005939 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(100.0 , 1 , "1").real()       , -0.006934 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(100.0 , 1 , "1").imag()       ,  0.000104 ,  eps);
+
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(0.55  , 2 , "0").real()       ,  1.611722 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(0.55  , 2 , "0").imag()       ,  0.079599 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(1.44  , 2 , "0").real()       ,  4.823117 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(1.44  , 2 , "0").imag()       ,  5.003168 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(1.65  , 2 , "0").real()       , -0.917640 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(1.65  , 2 , "0").imag()       ,  8.643233 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(100.0 , 2 , "0").real()       , -0.016038 ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_factor(100.0 , 2 , "0").imag()       ,  0.0      ,  eps);
+
+                double sp = 4.0 * 0.496 * 0.496;
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_outer_function(-100000.0 , sp , 0.0 , 4096 , 1 , "1").real(),  0.0      ,  eps);
+                TEST_CHECK_NEARLY_EQUAL( amp.omnes_outer_function(      0.0 , sp , 0.0 , 4096 , 1 , "1").real(),  0.26187  ,  eps);
+            }
```

Comment:
To diagnose the issue of the "unregistered" parameters, can we add a test case here where the parameter values are changed and the scattering amplitude is checked again?